frontend/src/routes/Governance/common/util.tsx

```diff
@@ -463,28 +464,69 @@ export function getPolicySetPolicies(policies: Policy[], policySet: PolicySet) {
   )
 }
 
-export function getPolicyRemediation(policy: Policy | undefined) {
+export function getPolicyRemediation(policy: Policy | undefined, propagatedPolicies: Policy[]) {
   if (!policy) {
     return ''
   }
-  if (policy.spec?.remediationAction) {
-    return policy.spec.remediationAction
-  }
   const templates = policy.spec['policy-templates']
-  let remediationAggregation = '-'
-  templates?.forEach((template: PolicyTemplate) => {
-    const templateRemediation = template.objectDefinition.spec.remediationAction
-    if (remediationAggregation === 'inform/enforce') {
-      return
-    } else if (remediationAggregation !== '-' && remediationAggregation !== templateRemediation) {
-      remediationAggregation = 'inform/enforce'
+  let rootRA = policy.spec.remediationAction
+  if (rootRA === REMEDIATION_ACTION.ENFORCE) {
+    return rootRA
+  }
+
+  // NOTE RA means Remediation Action
+  if (!rootRA && templates) {
+    for (const template of templates) {
+      const templateRemediation = template.objectDefinition.spec.remediationAction
+      if (!templateRemediation) {
+        continue
+      }
+      if (rootRA === REMEDIATION_ACTION.INFORM_ENFORCE) {
+        continue
+      }
+      if (!rootRA) {
+        rootRA = templateRemediation
+      } else if (rootRA !== templateRemediation) {
+        rootRA = REMEDIATION_ACTION.INFORM_ENFORCE
+        break
+      }
+    }
+  }
+
+  let remediationOverride = ''
+  let allOverridden = propagatedPolicies.length !== 0
+  propagatedPolicies.forEach((propaPolicy) => {
+    const propaPolicyRA = propaPolicy.spec.remediationAction
+    if (!propaPolicyRA) {
+      allOverridden = false
       return
-    } else if (remediationAggregation !== templateRemediation && templateRemediation) {
-      remediationAggregation = templateRemediation
+    }
+    if (propaPolicyRA === rootRA) {
+      allOverridden = false
       return
     }
+
+    if (!remediationOverride) {
+      remediationOverride = REMEDIATION_ACTION.ENFORCE
+    }
   })
-  return remediationAggregation
+
+  if (allOverridden) {
+    return REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
+  }
+
+  // No overrides are present
+  if (!remediationOverride) {
+    return rootRA
+  }
+
+  return REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDDEN
+}
+
+export function getObjectTempRemediation(propagatedPolicy: Policy, template: PolicyTemplate | undefined): string {
+  const propagatedRA = propagatedPolicy.spec.remediationAction
+  const objectRemediation = template?.objectDefinition.spec.remediationAction
+  return propagatedRA || objectRemediation || REMEDIATION_ACTION.INFORM
```

Comment:
It doesn't actually default to `inform`. It's possible that the user forgot to add the remediationAction field on the policy template and the policy which will eventually cause an error.  In this case, leaving it blank is correct.

Reply:
Thanks! great!